src/bucket/Bucket.cpp

```diff
@@ -961,11 +975,6 @@ mergeCasesWithEqualKeys(MergeCounters& mc, BucketInputIterator& oi,
         else
         {
             // Merge a create+delete to nothingness.
-            if (newEntry.type() != DEADENTRY)
```

Comment:
Just to sanity check, this code in inaccessible, right? I'm deleting it because I was thinking about if a valid `TEMPORARY` entry could somehow cause this to throw when I realized this branch is not possible.